examples/platforms/da15000/radio.c

```diff
@@ -0,0 +1,562 @@
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    (void)aInstance;
+    return kRadioCapsNone;
+}
+
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+    uint8_t value;
+    value = aEnable;
+    FTDF_setValue(FTDF_PIB_PROMISCUOUS_MODE, &value);
+    s_promiscuous = aEnable;
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+    return s_promiscuous;
+}
+
+ThreadError otPlatRadioHandleTransmitDone(bool *rxPending)
+{
+    ThreadError error = kThreadError_None;
+    VerifyOrExit(s_state == kStateTransmit, error = kThreadError_InvalidState);
+
+    s_state = kStateReceive;
+
+    if (rxPending != NULL)
+    {
+        *rxPending = s_data_pending;
+    }
+
+exit:
+    return error;
+}
+
+
+
+
+void FTDF_sendFrameTransparentConfirm(void *handle, FTDF_Bitmap32 status)
+{
+    (void)handle;
+    (void)status;
+    volatile uint32_t *txStatus = FTDF_GET_REG_ADDR_INDEXED(RETENTION_RAM_TX_RETURN_STATUS_1, FTDF_TX_DATA_BUFFER);
+
+    SendFrameDone = true;
+
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nTD", 3);
+#endif
+}
+
+
+
+void da15000RadioProcess(otInstance *aInstance)
+{
+    bool rxPending;
+    uint8_t bufferForFTDFconfigValue = 0;
+
+    if (SendFrameDone)
+    {
+        FTDF_SET_FIELD(ON_OFF_REGMAP_RXENABLE, 1);
+        otPlatRadioHandleTransmitDone(&rxPending);
+        otPlatRadioTransmitDone(aInstance, &s_transmit_frame, false, kThreadError_None);
+        SendFrameDone = false;
+    }
+
+    if (SED)
+    {
+        if ((otPlatAlarmGetNow() - sleep_const_delay) > 100)
+        {
+            FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &bufferForFTDFconfigValue);
+            s_state = kStateSleep;
+            SED = false;
+        }
+
+    }
+}
+
+
+
+void FTDF_rcvFrameTransparent(FTDF_DataLength frameLength,
+                              FTDF_Octet     *frame,
+                              FTDF_Bitmap32   status,
+                              FTDF_LinkQuality lqi)
+{
+    (void)status;
+    RadioPacket otRadioPacket;
+
+#if DEBUG_LOG_ENABLE
+    char buff[10];
+    sprintf(buff, "\nlq:%d", lqi);
+    hw_uart_write_buffer(HW_UART1, buff, 5);
+#endif
+
+    otRadioPacket.mPower = kPhyInvalidRssi;
+    otRadioPacket.mLqi   = lqi;
+
+    if (s_receive_frame.mChannel == 0)
+    {
+        s_receive_frame.mChannel = DEFAULT_CHANNEL;
+    }
+
+    otRadioPacket.mChannel =  s_receive_frame.mChannel;
+    otRadioPacket.mLength = frameLength;
+    otRadioPacket.mPsdu = frame;
+
+    otPlatRadioReceiveDone(First_Instace, &otRadioPacket, kThreadError_None);
+
+    if (s_state != kStateDisabled)
+    {
+        s_state = kStateReceive;
+    }
+
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    (void)aInstance;
+    aIeeeEui64[0] = 0x80;   //80-EA-CA is for Dialog Semiconductor
+    aIeeeEui64[1] = 0xEA;
+    aIeeeEui64[2] = 0xCA;
+    aIeeeEui64[3] = 0x00;
+    aIeeeEui64[4] = (NODE_ID >> 24) & 0xff;
```

Comment:
NODE_ID is currently statically set to 1, so every EUI-64 is going to be the same.  The EUI-64 needs to be unique.